### ISSUE TITLE
Adding a network bridge generates random binary altname

### ISSUE THREAD

**Issue by wsy2220** [systemd/systemd#15232]
**systemd version the issue has been seen with**
> systemd 245 (245.2-1)

**Used distribution**
 > Debian unstable with upstream kernel 5.5.13

**Expected behaviour you didn't see**
 > `ip link show br0` display no altname or meaningful altname

**Unexpected behaviour you saw**
 > `ip link show br0` display random binary altname

**Steps to reproduce the problem**
```
# ip link add br0 type bridge
# ip link show br0
4: br0: <BROADCAST,MULTICAST> mtu 1500 qdisc noop state DOWN mode DEFAULT group default qlen 1000
    link/ether e6:9c:fd:77:3c:e6 brd ff:ff:ff:ff:ff:ff
    altname �8Q�U
    altname �
             ��
```
With `80-net-setup-link.rules` masked this doesn't happen. But setting `net.ifnames=0` still show this. Sometimes I get 2 altnames and sometimes 1.

**Comment by poettering** [comment]
this looks like a bad bug.

**Comment by wsy2220** [comment]
I tried to build the debian package from source. Interestingly, the problem exists in default build, but disappeared with optimization off.

**Comment by poettering** [comment]
Pretty sure this is fixed by #15300. If not, please reopen.

**Comment by wsy2220** [comment]
Yes, it is fixed
